### PR TITLE
CDAP-2600 - Makes adapter name mandatory and checks for it before making backend call while creating adapter

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Make sure adapter name is entered before making a backend call.

### CODE REVIEW

File: cdap-ui/app/features/adapters/services/AdapterCreateModel.js [hunk 1]
```diff
@@ -105,22 +105,46 @@ angular.module(PKG.name + '.feature.adapters')
...
+          });
+          return defer.promise;
+        case 'invalidrequiredproperties':
           defer.reject({
             message: 'All required fields need to be set for all plugins.'
           });
           return defer.promise;
-        }
-        return formatAndSave.bind(this)();
+        case 'none':
```
Comment (elfenheart) [inline cdap-ui/app/features/adapters/services/AdapterCreateModel.js]: should this case be default instead?

File: cdap-ui/app/features/adapters/services/AdapterCreateModel.js [hunk 2]
```diff
@@ -105,22 +105,46 @@ angular.module(PKG.name + '.feature.adapters')
...
+        errObj.error = 'name';
+      } else if (this.source.placeHolder || this.sink.placeHolder) {
+        errObj.error = 'missingsourceorsink';
+      } else if (!this.validateRequiredProperties()) {
+        errObj.error = 'invalidrequiredproperties';
+      } else {
+        errObj.error = 'none';
+      }
+      return errObj;
+    }
```
Comment (elfenheart) [inline cdap-ui/app/features/adapters/services/AdapterCreateModel.js]: What happens when there are more than 1 error?

**Comment by ajainarayanan** [comment]
@elfenheart Fixed the issues mentioned. 
checkforname

**Comment by elfenheart** [comment]
:+1: LGTM

**Comment by elfenheart** [comment]
latest commit will conflict with #2868

**Comment by elfenheart** [comment]
LGTM

**Pull request merged**